app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt

```diff
@@ -342,6 +344,35 @@ enum class NavigationItem(
         override fun getRouteWithArgs(arguments: List<Any>): String = routeWithConversationIdArg(arguments)
     },
 
+    EditGuestAccess(
+        primaryRoute = EDIT_GUEST_ACCESS,
+        canonicalRoute = "$EDIT_GUEST_ACCESS?$EXTRA_CONVERSATION_ID={$EXTRA_CONVERSATION_ID}" +
+                "&$EXTRA_EDIT_GUEST_ACCESS_IS_GUEST_ACCESS_ALLOWED={$EXTRA_EDIT_GUEST_ACCESS_IS_GUEST_ACCESS_ALLOWED}" +
+                "&$EXTRA_EDIT_GUEST_ACCESS_IS_SERVICES_ALLOWED={$EXTRA_EDIT_GUEST_ACCESS_IS_SERVICES_ALLOWED}" +
+                "&$EXTRA_EDIT_GUEST_ACCESS_IS_UPDATING_GUEST_ACCESS_ALLOWED={$EXTRA_EDIT_GUEST_ACCESS_IS_UPDATING_GUEST_ACCESS_ALLOWED}",
+        content = { EditGuestAccessScreen() },
+        animationConfig = NavigationAnimationConfig.CustomAnimation(smoothSlideInFromRight(), smoothSlideOutFromLeft())
+    ) {
+        override fun getRouteWithArgs(arguments: List<Any>): String {
+            val conversationIdString: String = arguments.filterIsInstance<ConversationId>().firstOrNull()?.toString()
+                ?: "{$EXTRA_CONVERSATION_ID}"
+
+            val isGuestAccessAllowed: String = arguments.filterIsInstance<Boolean>().firstOrNull()?.toString()
+                ?: "{$EXTRA_EDIT_GUEST_ACCESS_IS_GUEST_ACCESS_ALLOWED}"
```

Comment:
yes correct it will crash if there is no guest value

Reply:
So it requires these params, we can ensure that we have them here, by either `checkNotNull` or `first` instead of `firstOrNull`, no need to try to open the screen if we know that it won't be successful 🤔